src/tools/compiletest/src/main.rs

```diff
@@ -32,6 +50,7 @@ help: try setting `profiler = true` in the `[build]` section of `config.toml`"#
         );
     }
 
+    warn_invalid_files(&config);
```

Comment:
Suggestion: so I dug through bootstrap test dispatching and compiletest, and it seems that for our purposes, it is correct and sufficient to put the warning inside compiletest (and not bootstrap). However, we should put it *after* all tests were attempted to be run, because compiletest has some convoluted test filtering / discovery / collection / execution logic, and at least to me the only reliable place to put it is after test run statistics were finalized and that we conclusively ran 0 *total* tests, i.e. see https://github.com/rust-lang/rust/commit/310165a9271aca14d6b9955d79257f8b1701f95e.

```rs
if st.total == 0 {
    panic!("no tests were run for the given filters: {:?}\n", opts.filters);
}
```

(I would like to tidy up / untangle compiletest a bit, but I never got around to it)

Reply:
Thank you for the detailed response.

Would it be better to check for `st.total < opts.filters.len() ` so we can catch cases like `./x test tests/ui/bad-name.rs tests/ui/bad-match.stderr` too?

Reply:
Hm. There *may* be edge cases where having less total tests ran than filters is reasonable or even desirable (but maybe I'm overthinking it), so maybe let's only do the exactly zero total test ran check to be conservative for now.

My thinking is that I really don't like that we have to do the check here to be reliable (the code paths involved in test discovery -> collection -> execution is extremely convoluted), and a more reliable check may be possible if there was a cleaner "phase separation" between the steps).

Reply:
Also note that my example was a panic not a warning, which is not the same as ur previous changes.

And that we should probably also check that filters is non-empty.

Reply:
Ah sure, given that this is not a reliable place to do the check, it's better to check for zero total tests. Also, updated the warning to a explicit panic.

Thank you for the detailed suggestions.